drivers/include/lcd.h

```diff
@@ -104,11 +161,15 @@ typedef struct lcd_driver lcd_driver_t;
  * @brief   Device descriptor for a lcd
  */
 typedef struct {
-#ifdef MODULE_DISP_DEV
+#if MODULE_DISP_DEV || DOXYGEN
     disp_dev_t *dev;                /**< Pointer to the generic display device */
 #endif
     const lcd_driver_t *driver;     /**< LCD driver */
     const lcd_params_t *params;     /**< Device initialization parameters */
+#if MODULE_LCD_PARALLEL || DOXYGEN
+    mutex_t lock;                   /**< Mutex used to lock the device in
```

Comment:
Good question :thinking: I have already discussed this with @aabadie and @bergzand and we basically agreed that mutual exclusion should be handled by the high-level driver, e.g. `disp_dev`. However, since `disp_dev` doesn't do this and the device driver implicitly uses mutual exclusion in SPI mode, I decided to implement mutual exclusion by the driver in parallel mode as well.

Reply:
I don't see a use case where we would ever have multiple threads access the low-level driver, so I'd just avoid the added complexity.

Reply:
But please document explicitly, that it is not thread safe. (I do agree with delegation of thread safety to upper layers - if needed at all. But being upfront about that may prevent long debugging sessions.)

Reply:
Hm, I'm not really sure. The execution of a function like `lcd_fill` at low level looks like this:
```c
    lcd_ll_acquire(dev);
    lcd_ll_set_area(dev, x1, x2, y1, y2);
    lcd_ll_cmd_start(dev, LCD_CMD_RAMWR, true);
    for (int i = 0; i < (num_pix - 1); i++) {
        lcd_ll_write_bytes(dev, true, (uint8_t *)&color, sizeof(color));
    }
    lcd_ll_write_bytes(dev, false, (uint8_t *)&color, sizeof(color));
    lcd_ll_release(dev);
```
That is we always have acquire the device execute some low-level transfers and release the device. This is necessary in SPI mode to avoid that each low-level transfer acquires and release the SPI device.

But what does `lcd_ll_acquire` and `lcd_ll_release` mean in parallel mode if we don't use a mutex? `lcd_ll_acquire(dev)` and `lcd_ll_release(dev)` are not internal function but part of the documented low-level API and suggest that the driver realizes mutual exclusion.